### ISSUE TITLE
Support ligatures.

### ISSUE THREAD

**Issue by jordwalke** [rogual/neovim-dot-app#107]
Some fonts, such as https://github.com/tonsky/FiraCode have ligatures and it would be great to support them.

**Comment by davidosomething** [comment]
+1

**Comment by philplckthun** [comment]
Me and @rogual are currently looking into rendering text more efficiently with CoreText. This would essentially support ligatures as well, I suppose. But we should keep in mind that the current rendering is very similar to iTerm's, which also doesn't support ligatures (?).

**Comment by jordwalke** [comment]
@philplckthun, Correct - I'm suggesting a way that neovim-dot-app could distinguish itself by offering features that other implementations *don't* support.

**Comment by philplckthun** [comment]
The commit I referenced modifies my PR to render characters together, thus allowing ligatures.

Check out PR #109 for more info.

**Comment by philplckthun** [comment]
@rogual Can be closed?

**Comment by bambu** [comment]
Agreed. I have been using FiraCode + Neovim.app since #109 was merged and it seems to work nicely.

**Comment by rogual** [comment]
@philplckthun No, I still need to test this. Just want to make sure ligatures work reliably, and not the sort-of-working that RTL text used to have.

**Comment by jordwalke** [comment]
I'm not exactly using the latest version, but often, when I type characters the ligatures don't draw. I have to highlight them with visual selection for them to rerender properly.

**Comment by philplckthun** [comment]
@jordwalke When Neovim/Vim draws new characters it draws them separately - naturally. Thus they are also rendered separately and the ligatures aren't drawn immediately. They only show up when the line is drawn as a whole.

This can't be fixed easily. We'd need a highly complex rendering system to circumvent these problems.

**Comment by rogual** [comment]
Then this is a bug, and does not constitute support for ligatures.

**Comment by jordwalke** [comment]
My suggestion in another neovim frontend was to keep track of drawn line content and always redraw individual lines as one unit using CoreText (or whatever). That would ensure that ligatures are always rendered correctly.

**Comment by rogual** [comment]
Yeah, sounds like the way to go.

**Comment by jamischarles** [comment]
Ligatures seem to be turned on for me (I'm using Menlo). It's causing some scroll position issues. Is there a way to turn this off?

**Comment by rogual** [comment]
No, it's a bug. OSX is being clever and sometimes rendering ligatures when we happen to draw several chars at the same time.
